### DISCUSSION TITLE
SEGY read implementation

### DISCUSSION THREAD

**Discussion by d-chambers** [DASDAE/dascore#261] [General]
This discussion serves as a place for me and @aaronjgirard to discuss details about the SEGY parser implementation.

First topic: getting the current SEGY test file from the Conoco Philips dataset:

```python
from dascore.utils.downloader import fetch

# download and locally cache segy file
path = fetch("conoco_segy_1.sgy")
```

**Comment by aaronjgirard** [comment]
There are several other python based segy readers that may be helpful here. I don't think we want to install anything else, but is that a place we can borrow from?

**Reply by d-chambers** [reply]
SEGY-SAK is a library with a nice API but they have a stronger licence than DASCore (GPL vs LGPL) so we can't use them directly as a dependency (from what I understand). I think, at least at this point, [segyio](https://github.com/equinor/segyio) would be a reasonable dependency though.

**Comment by aaronjgirard** [comment]
`from dascore.utils.downloader import fetch`

`Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/agirard/dascore/dascore/utils/downloader.py", line 23, in <module>
    fetcher.load_registry(REGISTRY_PATH)
  File "/home/agirard/miniconda2/envs/dascore/lib/python3.10/site-packages/pooch/core.py", line 648, in load_registry
    fin = stack.enter_context(open(fname))
FileNotFoundError: [Errno 2] No such file or directory: '/home/agirard/dascore/data_registry.txt'`

**Comment by d-chambers** [comment]
closed by #270
